src/file.cr

```diff
@@ -380,6 +380,62 @@ class File < IO::FileDescriptor
     end
   end
 
+  # Matches *path* against *pattern*.
+  #
+  # The pattern syntax is similar to shell filename globbing. It may contain the following metacharacters:
+  #
+  # * `*` matches an unlimited number of arbitrary characters.
+  #   * `"*"` matches all regular files.
+  #   * `"c*"` matches all files beginning with `c`.
+  #   * `"*c"` matches all files ending with `c`.
+  #   * `"*c*"` matches all files that have `c` in them (including at the beginning or end).
+  # * `?` matches any one charachter.
+  def self.fnmatch(pattern, path)
```

Comment:
Can we rename this to just `match`? `fnmatch` sounds super cryptic. I know it comes from C, but I don't think keeping C names is good. For example there's no `fnmatch` in Windows. So a name that's descriptive and applicable independent of the platform is preferred.

Also, please add `String` type restrictions here.